utils/base-images/Dockerfile

```diff
@@ -0,0 +1,54 @@
+# MIT License
+
+# Copyright (c) 2022 EASE lab
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# #---------- python-slim -----------#
+# # First stage (Builder):
+
+# FROM vhiveease/python-slim:latest as
+
+
+
+
+#---------- golang-builder -----------#
+FROM golang:1.16 AS golang-builder
+LABEL maintainer="https://github.com/ease-lab"
+
+
+#---------- python-builder -----------#
+FROM  python:3.8-slim AS python-slim
```

Comment:
here and everywhere: shouldn't it be `vhiveease/python-slim:latest` instead of `python-slim`?

Reply:
In the other benchmarks we only use the image name in the [Dockerfile](https://github.com/ease-lab/vSwarm/blob/590f7dda9f25b3d636c2dfc863062f6be0f461a3/benchmarks/video-analytics/Dockerfile#L23) then we tag it in the [Makefile](https://github.com/ease-lab/vSwarm/blob/590f7dda9f25b3d636c2dfc863062f6be0f461a3/benchmarks/video-analytics/Makefile#L35)
Therefore I did here the same to be consistent

Reply:
right but in that exact Dockerfile you mentioned, we use:
```
FROM vhiveease/python-slim:latest as recogBuilder
```

Reply:
No does not work.
The image name cannot contain symbols. We need use the tagging for it
![image](https://user-images.githubusercontent.com/31178749/183049518-79560fbd-fdec-4fbd-b779-f59c1aa65c50.png)

Reply:
I don't get it. how it work in one dockerfile and doesn't work in another one? I think you are not doint something right.

we need to have the base image's version fixed:
`FROM vhiveease/python-slim:latest as whatever`

instead of `python:3.8-slim` so that we can change the base to python:3.9-slim or 3.10, etc without changing the files, just by pushing a new image to `vhiveease/python-slim:latest` and `vhiveease/python-slim:3.9`

Reply:
I don't get it either. Isnt that exactly what we do here or not? generating the base image.
In this build we define on which upstream image we base our python-slim image on.

So what this build does it basically just retag the python:3.8-slim -> vhiveease/python-slim:latest
If we want to update the version we only need to change the FROM image (i.e. to 3.9) and then from now on all functions would be using python:3.9-slim as the base image.

Reply:
Sorry I was confused. It's fine

Reply:
No worries :)
I was a bit confused as well :)